core/src/main/java/com/linecorp/armeria/server/HttpServerHandler.java

```diff
@@ -254,11 +254,24 @@ private void handleRequest(ChannelHandlerContext ctx, DecodedHttpRequest req) th
         final String hostname = hostname(ctx, headers);
         final VirtualHost host = config.findVirtualHost(hostname);
 
+        final PathMappingContext mappingCtx = DefaultPathMappingContext.of(host, hostname,
+                                                                           path, query, headers,
+                                                                           host.producibleMediaTypes());
         // Find the service that matches the path.
-        final PathMapped<ServiceConfig> mapped = host.findServiceConfig(path, query);
+        final PathMapped<ServiceConfig> mapped;
+        try {
+            mapped = host.findServiceConfig(mappingCtx);
+        } catch (HttpResponseException cause) {
+            respond(ctx, req, cause.httpStatus());
+            return;
+        } catch (Throwable cause) {
+            logger.warn("Unexpected exception: {}", req, cause);
```

Comment:
`"{} Unexpected exception: {}", ctx.channel(), req, cause` so the connection information is also logged